tasks/runtime.yml

```diff
@@ -20,23 +20,22 @@
     mode: 0755
     state: directory
 
-- name: Generate a Munge key for the platform
+- name: Generate a Munge key
+  # NB this is usually a no-op as the package install actually generates a (node-unique) one, so won't usually trigger handler
   command: "dd if=/dev/urandom of=/etc/munge/munge.key bs=1 count=1024"
   args:
     creates: "/etc/munge/munge.key"
   when: inventory_hostname == openhpc_slurm_control_host
-  notify:
-    - Restart Munge service
 
-- name: Retrieve Munge key from Slurm control host
+- name: Retrieve Munge key
   slurp:
```

Comment:
Think this will read the file multiple times (once per host), but maybe that isn't too big of a deal if it simplifies the logic. Otherwise, you could keep the previous logic and do something like:
```
copy:
    content: "{{ omit if openhpc_munge_key_path else hostvars[openhpc_slurm_control_host]['slurm_munge_key']['content'] | b64decode }}"
    src: "{{ omit if not openhpc_munge_key_path else openhpc_munge_key_path }}"  
```

Reply:
It will but I think it's the simpliest option.

Reply:
fair enough